src/core/server/saved_objects/service/lib/errors.ts

```diff
@@ -202,4 +202,23 @@ export class SavedObjectsErrorHelpers {
   public static isGeneralError(error: Error | DecoratedError) {
     return isSavedObjectsClientError(error) && error[code] === CODE_GENERAL_ERROR;
   }
+
+  public static createGenericNotFoundEsUnavailableError(
+    type: string | null = null,
+    id: string | null = null
+  ) {
+    const notFoundError = this.createGenericNotFoundError(type, id);
+    return this.decorateEsUnavailableError(
+      new Error(`${notFoundError.message}`),
+      `x-elastic-product not present or not recognized`
```

Comment:
I'm not sure this is the best text string to use as a descriptor for the "missing" header. I took [inspiration](https://github.com/elastic/elasticsearch-js/blob/master/lib/errors.js#L141) from the client's team but made it a lot more specific. We need to get Product's input on wording if we choose to follow this implementation.